pkg/eth/tx_listener.go

```diff
@@ -29,13 +30,16 @@ const (
 
 // TxListener provides methods to interact with Ethereum transactions
 type TxListener struct {
-	blockchain ethereum.TransactionReader
-	jobs       jobs.JobService
+	blockchain    ethereum.TransactionReader
+	jobs          jobs.JobService
+	timeout       time.Duration
+	timeoutPassed bool
+	mutex         *sync.Mutex
 }
 
 // NewTxListener creates a new TransactionService instance
-func NewTxListener(blockchain ethereum.TransactionReader, jobs jobs.JobService) *TxListener {
-	return &TxListener{blockchain, jobs}
+func NewTxListener(blockchain ethereum.TransactionReader, jobs jobs.JobService, timeout time.Duration) *TxListener {
```

Comment:
this is going to break the API for clients. can you keep this function signature the same, and add `NewTxListenerWithTimeout`?